@nx/vite:test impossible to use a custom config file located inside a project root

### Current Behavior

When using @nx/vite:test you can specify a custom file using the `config` option.

Seems like there is a bug when the custom config file is located at the root of the project:

Let's assume we are in a lib called `my-test` in a workspace located at `/User/Marie/Dev` 
The custom config file is located in `/User/Marie/Dev/libs/my-test/vite.config.integration.ts`

If you provide a custom config file like: 

```
"test:integration": {
      "executor": "@nx/vite:test",
      ...
      "options": {
        "config": "vite.config.integration.ts",
      }
    }
```

The "resolved" config file path passed to vitest will be `/User/Marie/Dev/vite.config.integration.ts`
Note that the resolved path is not relative to the lib root but to the workspace root.

The [startVitest function is called](https://github.com/nrwl/nx/blob/e8c31d7ac72a6eeb98d07b61f6ae945a2612d8ac/packages/vite/src/executors/test/vitest.impl.ts#L62) with the following (partial) attributes:

```
{
  config: 'vite.config.integration.ts',
  root: 'libs/my-test',
}
```

So to fix that you'll want to use this path instead:

```
"test:integration": {
      "executor": "@nx/vite:test",
      ...
      "options": {
        "config": "libs/my-test/vite.config.integration.ts",
      }
    }
```

The [startVitest function is called](https://github.com/nrwl/nx/blob/e8c31d7ac72a6eeb98d07b61f6ae945a2612d8ac/packages/vite/src/executors/test/vitest.impl.ts#L62) with the following (partial) attributes:

```
{
  config: 'libs/my-test/vite.config.integration.ts',
  root: 'libs/my-test',
}
```

And since vitest seems to be executed using a PWD equal to the workspace root, it will end up trying to load the following config file:

`/User/Marie/Dev/libs/my-test/libs/my-test/vite.config.integration.ts` which does not exist.






### Expected Behavior

Specifying a custom config file should work no matter where it's located.

### GitHub Repo

_No response_

### Steps to Reproduce

1. Use a "@nx/vite:test" executor in one of your lib
2. Create a custom vitest config file a the root of your lib
3. try to reference it using the config option


### Nx Report

```shell
Node   : 16.17.1
   OS     : darwin arm64
   yarn   : 1.22.19
   Hasher : Native
   
   nx                 : 16.0.3
   @nx/js             : 16.0.3
   @nx/jest           : 16.0.3
   @nx/linter         : 16.0.3
   @nx/workspace      : 16.0.3
   @nx/devkit         : 16.0.3
   @nx/esbuild        : 16.0.3
   @nx/eslint-plugin  : 16.0.3
   @nx/node           : 16.0.3
   @nx/plugin         : 16.0.3
   @nx/react          : 16.0.3
   @nrwl/tao          : 16.0.3
   @nx/vite           : 16.0.3
   @nx/web            : 16.0.3
   typescript         : 4.9.5
```


### Failure Logs

```shell
✘ [ERROR] Could not resolve "/User/Marie/Dev/libs/my-test/libs/my-test/vite.config.integration.ts"
failed to load config from /User/Marie/Dev/libs/my-test/libs/my-test/vite.config.integration.ts
```


### Operating System

- [X] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

IMHO the fix would be to path an absolute path to the startVitest function:

Adding `config: viteConfigPath,` around the line 113 of packages/vite/src/executors/test/vitest.impl.ts solved the issue for me.

--- a/packages/vite/src/executors/test/vitest.impl.ts
+++ b/packages/vite/src/executors/test/vitest.impl.ts
@@ -53,7 +53,7 @@ export async function* vitestExecutor(
 ) {
   const projectRoot =
     context.projectsConfigurations.projects[context.projectName].root;
-  registerTsConfigPaths(resolve(projectRoot, 'tsconfig.json'));
+  registerTsConfigPaths(resolve(workspaceRoot, projectRoot, 'tsconfig.json'));
 
   const { startVitest } = await (Function(
     'return import("vitest/node")'
@@ -111,15 +111,23 @@ async function getSettings(
     : ({} as CoverageOptions);
 
   const viteConfigPath = options.config
-    ? joinPathFragments(context.root, options.config)
+    ? options.config // config is expected to be from the workspace root
     : findViteConfig(joinPathFragments(context.root, projectRoot));
+
+  const resolvedProjectRoot = resolve(workspaceRoot, projectRoot);
+  const resolvedViteConfigPath = resolve(
+    workspaceRoot,
+    projectRoot,
+    relative(resolvedProjectRoot, viteConfigPath)
+  );
 
   const resolved = await loadConfigFromFile(
     {
       mode: options.mode,
       command: 'serve',
     },
-    viteConfigPath
+    resolvedViteConfigPath,
+    resolvedProjectRoot
   );
 
   if (!viteConfigPath || !resolved?.config?.['test']) {
@@ -138,7 +146,8 @@ You can manually set the config in the project, ${
     // when running nx from the project root, the root will get appended to the cwd.
     // creating an invalid path and no tests will be found.
     // instead if we are not at the root, let the cwd be root.
-    root: offset === '' ? projectRoot : '',
+    root: offset === '' ? resolvedProjectRoot : workspaceRoot,
+    config: resolvedViteConfigPath,
     reporters: [
       ...(options.reporters ?? []),
       ...((resolved?.config?.['test']?.reporters as string[]) ?? []),

